homeassistant/components/shelly/config_flow.py

```diff
@@ -182,3 +186,42 @@ async def _async_get_info(self, host):
                 aiohttp_client.async_get_clientsession(self.hass),
                 host,
             )
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Define the config flow to handle options."""
+        return OptionsFlow(config_entry)
+
+
+class OptionsFlow(config_entries.OptionsFlow):
+    """Handle options for Shelly."""
+
+    def __init__(self, config_entry):
+        """Initialize."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        return self.async_show_form(
+            step_id="init",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
```

Comment:
Why make it two options? If someone uses it for something else, they don't want any of the light capabilities and instead want the dimmer to be shown as a switch?

Reply:
I thought about that too, but then also considered the case where users would use white dimming bulbs (color not supported). I would imagine that it's a relatively common use case?

Reply:
I suggest you start with just having it allow picking between switch and light. You can make it pick what to set up inside `shelly/light.py` method `async_setup_entry`

Reply:
Sorry but I can't figure out how to handle only showing the options for light devices and not for the other types. I tried to get `async_get_options_flow` to return all sorts of different things, but I'm not managing. I assume you mean that from `async_setup_entry` in `shelly/light.py` I can init a device as a switch rather than a light, right? I'm okay with that, but how do I set up the options flow to give the user that option?

Reply:
Any help would be appreciated

Reply:
We might need a change to how Home Assistant works. We can look into that once the option is added to make it a light/switch platform.

Reply:
Oh wow! Okay, in the meantime, would you consider implementing it like I did, or would you rather wait?

Reply:
@balloob ⬆️

Reply:
We need to implement it the correct way, else we need to make a breaking change.

Reply:
Do we have an issue / PR we can track that work on?